eng/azure-pipelines-codeql.yml

```diff
@@ -0,0 +1,102 @@
+trigger:
+  none
+
+schedules:
+  - cron: 0 12 * * 1
+    displayName: Weekly Monday CodeQL/Semmle run
+    branches:
+      include:
+      - dotnet/main
```

Comment:
I guess that in the case we try upgrading llvm we would need to add that or any branch to this list, right?

Reply:
For most repos, CodeQL is only enabled for the default branch. From what I understand, it seems like it should be enabled for the branches that support a service/product. I'm not familar with the different branches, but @steveisok mentioned that codeql should be enabled for objwriter branch as well.
Would upgrading llvm lead to a completely new branch in this repository or a new default branch? And would it support a service/build that is different from dotnet/main and objwriter/12.x?

Reply:
It is not decided yet, but it may lead to a new branch, it would support the same service/builds as objwriter/12.x. A reason to keep them separated is having a stable version for a .net release while continuing upgrading for development/future .net releases.

Reply:
In the event that a new branch becomes the default for what objwriter/12.x (is based on llvm 12 right?) builds towards (is it a native AOT focused build?), I believe we could probably replace `objwriter/12.x` in this block with that name. Especially if the new branch is a superset of the changes that make it to the stable branch, which matches how other repos are using CodeQL on their main branch and is not enabled by default for other branches like their stable release branches.

I dont think we would *need* to continue running codeql on `objwriter/12.x` if it gets replaced, but I think it could still run if it adds value.